### DISCUSSION TITLE
prefix requirement for sources?

### DISCUSSION THREAD

**Discussion by abenz1267** [hrsh7th/nvim-cmp#1089] [Ideas]
Hi,

is it somehow possible to have a prefix-requirement for certain sources? F.e. i only want snippets when i prefix with "!". Would that somehow be doable with the current version?

Regards

**Comment by steliyan** [comment]
This sounds like a very useful feature.

Maybe it's possible to use the [`entry_filter`](https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt#L570).

Not sure how to properly get the `filter_text`, it seems the value is cached.
